src/navigation/navigation-events.js

```diff
@@ -117,21 +123,22 @@ window.removeEventListener = function(eventName, listenerFn) {
   return originalRemoveEventListener.apply(this, arguments);
 };
 
-const originalPushState = window.history.pushState;
-window.history.pushState = function(state) {
-  const result = originalPushState.apply(this, arguments);
+window.history.pushState = patchedUpdateState(window.history.pushState);
+window.history.replaceState = patchedUpdateState(window.history.replaceState);
 
-  urlReroute(createPopStateEvent(state));
+function patchedUpdateState(updateState) {
+  return function() {
+    const urlBefore = window.location.href;
+    const result = updateState.apply(this, arguments);
+    const urlAfter = window.location.href;
 
-  return result;
-};
+    if (!urlRerouteOnly || urlBefore !== urlAfter) {
```

Comment:
hmm, forgive me for being slow today, I'm having a hard time.

the way this is written here, could it potentially be a breaking change? because this is looking at either the new (optional) prop OR if the url before is not the same as the url after

whereas previously it would always fire even if the urls are the same?

again, sorry, I'm having a hard time today, so I could be reading that wrong.

Reply:
For all preexisting code, `urlRerouteOnly` will be `false`.

`!urlRerouteOnly || urlBefore !== urlAfter` ->`!false || urlBefore !== urlAfter` -> `true || urlBefore !== urlAfter` -> `true`

In other words, it's an `if (true)` for all pre-existing code

Reply:
haha yes, thanks. I knew I was having a hard time.